### PR TITLE
Remove build-time DOM dependencies from base tsconfigs

### PR DESCRIPTION
**Pull request by TommyBrosman**
- Removed DOM API dependencies from base tsconfigs
- Added an explicit dependency on DOM APIs in `property-properties`, which uses `console`.

### CODE REVIEW

**Comment by Josmithr** [comment]
@tylerbutler are we still publishing `build-common` and leveraging it as a package anywhere? Or are we strictly consuming it via file path these days? Wondering what more we may need to do to validate this / what additionally we may need to do after merging.

**Comment by tylerbutler** [comment]
> @tylerbutler are we still publishing `build-common` and leveraging it as a package anywhere? Or are we strictly consuming it via file path these days? Wondering what more we may need to do to validate this / what additionally we may need to do after merging.

It's still a dep in server as far as I know. It's hard to use relative paths there because server builds in docker.

**Comment by TommyBrosman** [comment]
I'm updating the API report for api-markdown-documenter. Originally there were import statements that aliased Node as Node_2:

```
…
-import type { Node as Node_2 } from 'unist';
+import type { Node } from 'unist';

…
-export interface DocumentationNode<TData extends object = Data> extends Node_2<TData> {
+export interface DocumentationNode<TData extends object = Data> extends Node<TData> {
```

I believe this is because [Node](https://developer.mozilla.org/en-US/docs/Web/API/Node) is a DOM type in the global scope when "DOM" is specified as a lib in the tsconfig, and there is yet another `Node` type coming from `unist`.

**Review by Josmithr — APPROVED** [review]

File: build-tools/packages/build-cli/tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
 	"include": ["src/**/*"],
 	"exclude": ["src/test/**/*"],
 	"compilerOptions": {
+		"lib": ["DOM"],
```
[the remark this answers is not in the document]
Comment (TommyBrosman) [reply build-tools/packages/build-cli/tsconfig.json:6]: Ah, does it not merge with the libs from the base tsconfig?

[the remark this answers is not in the document]
Comment (alexvy86) [reply build-tools/packages/build-cli/tsconfig.json:6]: As far as I could tell, no. The whole array is overridden.

**Comment by TommyBrosman** [comment]
I'm abandoning this PR in favor of a simpler PR that changes the base tsconfigs without eliminating the dependency. After that goes in, it will be easier to eliminate usages from individual packages.

**Pull request closed without merging**